plugins/jetpants_collins/topology.rb

```diff
@@ -274,12 +313,43 @@ def query_spare_assets(count, options={})
       # Now iterate in a single-threaded way for simplicity
       nodes.each do |node|
         db = node.to_db
-        if(db.usable_spare? && (!source || db.usable_with?(source)))
-          keep_nodes << node
-          break if keep_nodes.size >= count
+        if(db.usable_spare? &&
+          (
+            !source ||
+            (!source.pool && db.usable_with?(source)) ||
+            (
+              (!options[:for_pool] && source.pool && db.usable_in?(source.pool)) ||
+              (options[:for_pool] && db.usable_in?(options[:for_pool]))
```

Comment:
@KirilAngov it appears so, but in truth this conditional is a bit messy and could be broken up for clarity